components/datetime/src/any/datetime.rs

```diff
@@ -203,34 +220,31 @@ impl AnyDateTimeFormat {
     }
     /// Takes a [`DateTimeInput`] implementer and returns an instance of a [`FormattedDateTime`]
     /// that contains all information necessary to display a formatted date and operate on it.
-    ///
-    /// This function will fail if the date passed in uses a different calendar than that of the
-    /// AnyCalendar. Please convert dates before passing them in if necessary.
     #[inline]
-    pub fn format<'l, T>(
-        &'l self,
-        value: &'l T,
-    ) -> Result<FormattedDateTime<'l>, DateTimeFormatError>
+    pub fn format<'l, T>(&'l self, value: &T) -> FormattedDateTime<'l>
```

Comment:
Why did you change this behavior from being fallible to auto-converting?

Reply:
That's what we had discussed, no?

Reply:
In the previous PR you had asked for that error mode to be removed and I had agreed to do that in a followup.

Reply:
Oh, that was for a different error. Either way, I was under the impression we wanted this to be autoconverting. If not, I can revert.